clippy_lints/src/literal_representation.rs

```diff
@@ -414,20 +414,27 @@ impl LiteralDigitGrouping {
         radix: Radix,
         lint_unreadable: bool,
     ) -> Result<Option<usize>, WarningType> {
-        let mut groups = groups.map(str::len);
-
-        let first = groups.next().expect("At least one group");
+        let groups = groups.collect::<Vec<_>>();
+        let Some(first) = groups.first().map(|group| group.len()) else {
+            return Ok(None);
+        };
 
-        if radix == Radix::Binary || radix == Radix::Octal || radix == Radix::Hexadecimal {
-            if let Some(second_size) = groups.next() {
-                if !groups.all(|i| i == second_size) || first > second_size {
-                    return Err(WarningType::UnusualByteGroupings);
+        if groups.len() > 1 && (radix == Radix::Binary || radix == Radix::Octal || radix == Radix::Hexadecimal) {
+            if let [part_1, part_2, part_3, part_4] = groups.as_slice() {
+                if part_1.len() == 3 && part_2.len() == 2 && part_3.len() == 2 && part_4.len() == 1 {
```

Comment:
Hmmm... This would indeed fix #11749, but the false positive would still be there if the literal is e.g. `0b0_11_00_0000`.

I'm not sure how to tackle this :thinking:

Reply:
I can handle this case as well if you want?

Reply:
I don't know Guillaume, like, even if this change took into account that second scenario, and even a third one, it would just be putting a temporary fix over a systematic issue.

Maybe extracting binary checking as a configuration option would be better. But this is in general an impossible problem (making Clippy comprehend intent?)